cli/flags.rs

```diff
@@ -139,6 +139,9 @@ pub struct Flags {
   pub allow_write: Option<Vec<PathBuf>>,
   pub ca_file: Option<String>,
   pub cache_blocklist: Vec<String>,
+  /// This is not exposed as an option in the CLI, it is used internally when
+  /// the language server is configured with an explicit cache option.
+  pub cache_path: Option<PathBuf>,
```

Comment:
I also wish we didn't have to do this, but I don't know any other suggestion other than creating a new struct with specific options for the `ProgramState`, which might be more trouble than it's worth.

Reply:
Yeah, that or another argument to the new function, but that was even uglier.